### DISCUSSION TITLE
Search does not include nested collections

### DISCUSSION THREAD

**Discussion by jojomillennium** [directus/directus#3040] [Feature Requests]
9.0.0rc2 docker image with DB pointing on a mysql 5.7

Searching entries in a collection does not include nested elements.

![image](https://user-images.githubusercontent.com/6372094/97289755-e0e85800-1847-11eb-8454-ea3557e85799.png)

Typically, searching in a collection having a field of type translations, will not search in the content of the corresponding translations themselves as they are located in another collection.

![image](https://user-images.githubusercontent.com/6372094/97290184-813e7c80-1848-11eb-974b-9ed8c704813f.png)

**Comment by benhaynes** [comment]
Not sure how this would be done without initial configuration or a more dedicated service. Relational data can be deeply nested, and you can have _many_  parent/related items... so this query would be pretty complicated and slow (especially considering we now support M2A relationships). I imagine this would need to be configured to allow-list the searchable relational fields, or have some dedicated table for concatenated values that can be more efficiently searched. @rijkvanzanten — thoughts?

**Reply by vidhav** [reply]
From a users perspective, the translated values seem to be a part of the collection they are in. I can understand the complexity if you open it up for searching too deep in foreign values, but maybe there could be an exception for translations...

Maybe a Directus admin could set up a database view, with all the joins, and point the collection search to that view?

Table name: `my_table` => View name: `view_my_table_search`.

**Reply by rijkvanzanten** [reply]
> but maybe there could be an exception for translations...

Even with that, there will be a theoretical infinite level of recursion and nesting that can happen just on that one o2m to the translated value 🤔 I also think the same problem/feature requests exists for any regular m2o value, which widens this scope right back up to where we are now. I think your idea around relying on views for this purpose makes a lot of sense. Alternatively, we could look into ways to "upgrade" search using "search adapters" (similar to storage/auth) to allow for more in-depth searching algorithms to be applied (for example by integrating with elastic search or algolia for example).

**Reply by benhaynes** [reply]
I really like the option of modularizing search so there are options to upgrade it to elasticsearch, algolia, etc.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
